src/cls/rbd/cls_rbd.cc

```diff
@@ -5052,8 +5052,8 @@ int trash_list(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   CLS_LOG(20, "trash_get_images");
   while (data.size() < max_return) {
     map<string, bufferlist> raw_data;
-    int max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
-                                     max_return - data.size());
+    max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
+                                 max_return - data.size());
     int r = cls_cxx_map_get_vals(hctx, last_read, trash::IMAGE_KEY_PREFIX,
```

Comment:
I meant like this. There is no reason for the assignment, as it is done immediately after entering the while loop. The other branch `not entering the while loop` doesn't use `max_read` too.

```
int max_read;
bool more = true;

  CLS_LOG(20, "trash_get_images");
  while (data.size() < max_return) {
    map<string, bufferlist> raw_data;
    max_read = std::min<int32_t>(RBD_MAX_KEYS_READ,
                                 max_return - data.size());

```